tools/range_read/range_read.c

```diff
@@ -610,66 +643,62 @@ main(
         /* Start all the loaders in a detached state so we can have them
          * running while the exec phase is running too.
          */
-        stopthreads = false;
-        snprintf(logfile, sizeof(logfile), "rr_load_%u.log", opts.sfx_start);
-        kh_register(KH_FLAG_DETACH, &print_stats, logfile);
+        kh_register(0, &print_stats, 0);
         for (i = 0; i < opts.threads; i++)
-            kh_register_kvs(kvs, KH_FLAG_DETACH, &kvs_cparms, &kvs_oparms, &loader, &ti[i]);
+            kh_register_kvs(kvs, 0, &kvs_cparms, &kvs_oparms, &loader, &ti[i]);
 
-        while (!stopthreads && atomic_read(&n_write) < tot_keys)
+        while (atomic_read(&n_write) < tot_keys)
             sleep(5);
     }
 
     if (opts.phase & EXEC) {
-        long tot_mem;
-        uint warmup_nkeys;
-        uint tot_keys;
-        char logfile[2048];
+        struct lat_hist *lat;
 
-        stopthreads = false;
+        lat = malloc(opts.threads * sizeof(*lat));
```

Comment:
How big is *lat?  Maybe use aligned_alloc() here so that all the *lat objects are cacheline aligned?  (actually 2x cacheline aliagned)

Reply:
Ah, I see, they are HSE_ACP_ALIGNED (i.e., 2 cachelines).  You should call aligned_alloc(HSE_ACP_ALIGNED, sizeof(*lat) * opts.threads)...